core/services/workflows/artifacts/orm.go

```diff
@@ -252,11 +325,10 @@ func (orm *orm) UpsertWorkflowSpec(ctx context.Context, spec *job.WorkflowSpec)
 				:created_at,
 				:updated_at,
 				:spec_type
-			) ON CONFLICT (workflow_owner, workflow_name) DO UPDATE
+			) ON CONFLICT (workflow_id) DO UPDATE
 			SET
 				workflow = EXCLUDED.workflow,
 				config = EXCLUDED.config,
-				workflow_id = EXCLUDED.workflow_id,
 				workflow_owner = EXCLUDED.workflow_owner,
```

Comment:
I wonder if we should restrict these further. If the WorkflowID is the same you wouldn't expect any of the fields that go into generating it to change